source/solvers/tracer.cc

```diff
@@ -219,6 +400,138 @@ Tracer<dim>::assemble_system_rhs()
   this->system_rhs.compress(VectorOperation::add);
 }
 
+template <int dim>
+void
+Tracer<dim>::assemble_system_rhs_dg()
+{
+  const DoFHandler<dim> *dof_handler_fluid =
+    multiphysics->get_dof_handler(PhysicsID::fluid_dynamics);
+
+  auto scratch_data = TracerScratchData<dim>(
+    this->simulation_parameters.physical_properties_manager,
+    *this->fe,
+    *this->cell_quadrature,
+    *this->face_quadrature,
+    *this->mapping,
+    dof_handler_fluid->get_fe());
+
+  StabilizedDGMethodsCopyData copy_data(this->fe->n_dofs_per_cell(),
+                                        this->cell_quadrature->size());
+
+  const auto cell_worker =
+    [&](const typename DoFHandler<dim>::active_cell_iterator &cell,
+        TracerScratchData<dim>                               &scratch_data,
+        StabilizedDGMethodsCopyData                          &copy_data) {
+      this->assemble_local_system_rhs(cell, scratch_data, copy_data);
+    };
+
+  const auto boundary_worker =
+    [&](const typename DoFHandler<dim>::active_cell_iterator &cell,
+        const unsigned int                                   &face_no,
+        TracerScratchData<dim>                               &scratch_data,
+        StabilizedDGMethodsCopyData                          &copy_data) {
+      // Identify which boundary condition corresponds to the boundary id. If
+      // this boundary condition is not identified, then exit the simulation
+      // instead of assuming an outlet.
+      const auto &triangulation_boundary_id =
+        cell->face(face_no)->boundary_id();
+      const unsigned int boundary_index =
+        get_lethe_boundary_index(triangulation_boundary_id);
+
+      scratch_data.reinit_boundary_face(
+        cell,
+        face_no,
+        boundary_index,
+        this->evaluation_point,
+        this->multiphysics->get_immersed_solid_signed_distance_function());
+
+      // Gather velocity information at the face to properly advect
+      // First gather the dof handler for the fluid dynamics
+      const DoFHandler<dim> *dof_handler_fluid =
+        multiphysics->get_dof_handler(PhysicsID::fluid_dynamics);
+      // Identify the cell that corresponds to the fluid dynamics
+      typename DoFHandler<dim>::active_cell_iterator velocity_cell(
+        &(*triangulation), cell->level(), cell->index(), dof_handler_fluid);
+
+      reinit_face_velocity_with_adequate_solution(velocity_cell,
+                                                  face_no,
+                                                  scratch_data);
+
+      scratch_data.calculate_face_physical_properties();
```

Comment:
regrettably at the moment this would not be trivial. But I can try to fix that in the future. I don,t mind repeating 4 lines for the moment.

Reply:
This is also an issue we have with all RHS and Matrix assembly. Eventually what we could do would be to have a template argument to have the same function do RHS and matrix assembly. This would remove code duplication. I will open an issue on this.